spring-cloud-azure-samples/azure-appconfiguration-conversion-sample/README.md

```diff
@@ -0,0 +1,214 @@
+
+1. Use the Azure CLI [az keyvault create](https://docs.microsoft.com/en-us/cli/azure/cosmosdb?view=azure-cli-latest#az-cosmosdb-create)
+
+    ```azurecli
+    az appconfig create --name myConfigStoreName -g MyResourceGroup -l eastus
+    ```
+
+1. For this tutorial, you'll use a service principal for authentication to KeyVault. To create this service principal, use the Azure CLI [az ad sp create-for-rbac](/cli/azure/ad/sp?view=azure-cli-latest#az-ad-sp-create-for-rbac) command:
+
+    ```azurecli
+    az ad sp create-for-rbac -n "http://mySP" --sdk-auth
+    ```
+
+    This operation will return a series of key / value pairs.
+
+    ```console
+    {
+    "clientId": "iiiiiiii-iiii-iiii-iiii-iiiiiiiiiiii",
+    "clientSecret": "ssssssss-ssss-ssss-ssss-sssssssssss",
+    "subscriptionId": "bbbbbbb-bbbb-bbbb-bbbb-bbbbbbbbbbb",
+    "tenantId": "ttttttt-tttt-tttt-tttt-ttttttttttt",
+    "activeDirectoryEndpointUrl": "https://login.microsoftonline.com",
+    "resourceManagerEndpointUrl": "https://management.azure.com/",
+    "activeDirectoryGraphResourceId": "https://graph.windows.net/",
+    "sqlManagementEndpointUrl": "https://management.core.windows.net:8443/",
+    "galleryEndpointUrl": "https://gallery.azure.com/",
+    "managementEndpointUrl": "https://management.core.windows.net/"
+    }
+    ```
+
+1. Run the following command to allow the service principal to access your key vault:
+
+    ```azurecli
+        az keyvault set-policy -n myVaultName --spn "iiiiiiii-iiii-iiii-iiii-iiiiiiiiiiii" --secret-permissions get
+    ```
+
+1. Upload your Cosmos DB key to Key Vault.
+
+    ```azurecli
+        az keyvault secret set --vault-name myKeyCaultName --name "COSMOSDB-KEY" --value yourCosmosDBKey
+    ```
+
+1. Upload your Configurations Cosmos DB name and URI to App Configuration
+
+    ```azurecli
+        az appconfig kv set --name myConfigStoreName --key "/application/azure.cosmosdb.database" --value myCosmosDBName --content-type " " --yes
+        az appconfig kv set --name myConfigStoreName --key "/application/azure.cosmosdb.uri" --value myCosmosDBUri --content-type " " --yes
+    ```
+
+1. Add a Key Vault Reference to App Configuration, make sure to update the uri with your config store name.
+
+    ```azurecli
+        az appconfig kv set --name myConfigStoreName --key "/application/azure.cosmosdb.key" --value "{\"uri\":\"https://myConfigStoreName.vault.azure.net/secrets/COSMOSDB-KEY\"}" --content-type "application/vnd.microsoft.appconfig.keyvaultref+json;charset=utf-8" --yes
+    ```
+
+1. Get a connection string to App Configuration.
+
+    ```azurecli
+        az appconfig credential list -g MyResourceGroup --name myConfigStoreName
+    ```
+
+1. Delete `application.propertes` from `src/main/resources`.
+
+1. Create a new file called `bootstrap.properties` in `src/main/resources`, and add the following.
+
+    ```properties
+        spring.cloud.azure.appconfiguration.stores[0].connection-string=${CONFIG_STORE_CONNECTION_STRING}
```

Comment:
There needs to be a newline at the end of this line, otherwise `maven-checkstyle-plugin` will complain.

Reply:
Newline with no whitespace ;)